src/pallene/uninitialized.lua

```diff
@@ -156,31 +109,50 @@ function uninitialized.verify_variables(module)
 
         local nvars = #func.vars
         local nargs = #func.typ.arg_types
-        local nret  = #func.typ.ret_types
 
-        local falls_through
-        local analysis = coroutine.wrap(function()
-            local uninit = {}
-            for i = nargs+1, nvars do
-                uninit[i] = true
+        -- initialize sets
+        local kill_sets = {}
+        local uninit_sets = {}
+        for _,b in ipairs(func.blocks) do
+            local kill = gen_kill_set(b)
+            table.insert(kill_sets, kill)
+            table.insert(uninit_sets, {})
+        end
+        local entry_uninit = uninit_sets[1]
+        for v_i = nargs+1, nvars do
+            entry_uninit[v_i] = true
+        end
+
+        -- solve flow equations
+        flow_analysis(func.blocks, uninit_sets, kill_sets)
+
+        -- check for errors
+        local check = coroutine.wrap(function()
```

Comment:
We used to need the coroutine cleverness because the analysis was a recursive traversal. Is this still needed?

Reply:
No, I'll change it to make a simple loop.